src/test/java/edu/ucla/library/bucketeer/verticles/ClearCacheVerticleTest.java

```diff
@@ -0,0 +1,153 @@
+
+    /**
+    * A test context from which references to Vert.x can be retrieved.
+    */
+    @Rule
+    public RunTestOnContext myRunTestOnContextRule = new RunTestOnContext();
+
+    /**
+     * Cantaloupe username for testing Cantaloupe cache clearing.
+     */
+    private String myUsername;
+
+    /**
+     * Cantaloupe password for testing Cantaloupe cache clearing.
+     */
+    private String myPassword;
+
+    /**
+     * Verticle ID for undeploying ClearCacheVerticle
+     */
+    private String myVertID;
+
+    /**
+     * Sets up the tests.
+     *
+     * @param aContext A testing environment
+     */
+    @Before
+    public final void setup(final TestContext aContext) {
+        final ConfigRetriever configRetriever = ConfigRetriever.create(Vertx.vertx());
+        final Async asyncTask = aContext.async();
+
+        configRetriever.getConfig(configuration -> {
+            if (configuration.failed()) {
+                aContext.fail(configuration.cause());
+            } else {
+                final JsonObject config = configuration.result();
+                final DeploymentOptions options = new DeploymentOptions().setConfig(config);
+
+                myRunTestOnContextRule.vertx()
+                    .deployVerticle(ClearCacheVerticle.class.getName(), options, deployment -> {
+                        if (deployment.succeeded()) {
+                            myVertID = deployment.result();
+                            TestUtils.complete(asyncTask);
+                        } else {
+                            aContext.fail(deployment.cause());
+                        }
+
+                    });
+            }
+        });
+
+
+    }
+
+
+    /**
+     * Tear down the testing environment.
+     *
+     * @param aContext A test context
+     */
+    @After
+    public void tearDown(final TestContext aContext) {
+
+        if (myVertID != null) {
+            final Async async = aContext.async();
+            myRunTestOnContextRule.vertx().undeploy(myVertID, undeployment -> {
+                if (undeployment.failed()) {
+                    aContext.fail(undeployment.cause());
+                } else {
+                    async.complete();
+                }
+            });
+        } else {
+            LOGGER.info(MessageCodes.BUCKETEER_605);
+        }
+
+
+
+       //in tear down get verticle id and shut down the verticle by referencing the ID
+    }
+
+    /**
+     * Tests clearing Cantaloupe's cache.
+     *
+     * @param aContext A testing context
+     */
+    @Test
+    public void testCacheClear(final TestContext aContext) {
+
+        final WebClient client = WebClient.create(Vertx.vertx());
+
+        // These are just the property names, not values
+        LOGGER.info(MessageCodes.BUCKETEER_606, StringUtils.trimToNull(myUsername));
+
+        //Sends multiple messges over eventBus to see if correct response is recieved each time
+        for (int i = 0; i < 500; i++) {
```

Comment:
Another example of where we embrace Java's verbosity and eschew overly simple variable names (like "i", even when there is a long history in programming of this particular short-hand). Typically, here, we'd use:

`for (int index = 0; index < 500; index++)`

Reply:
Since someone might wonder why we do this when there is a single check too, you might want to add something to the comment about having seen different results for multiple queued requests in the past.